Harvest.Email/Services/EmailBodyService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Routing;
+
+namespace Harvest.Email.Services
+{
+    // Code from: https://github.com/aspnet/Entropy/blob/dev/samples/Mvc.RenderViewToString/RazorViewToStringRenderer.cs
+    public interface IEmailBodyService
+    {
+        Task<string> RenderBody<TModel>(string viewName, TModel model);
+    }
+
+    public class EmailBodyService : IEmailBodyService
+    {
+        private readonly IRazorViewEngine _viewEngine;
+        private readonly ITempDataProvider _tempDataProvider;
+        private readonly IServiceProvider _serviceProvider;
+
+        public EmailBodyService(
+            IRazorViewEngine viewEngine,
+            ITempDataProvider tempDataProvider,
+            IServiceProvider serviceProvider)
+        {
+            _viewEngine = viewEngine;
+            _tempDataProvider = tempDataProvider;
+            _serviceProvider = serviceProvider;
+        }
+
+        public async Task<string> RenderBody<TModel>(string viewName, TModel model)
+        {
+            var actionContext = GetActionContext();
+            var view = FindView(actionContext, viewName);
+
+            await using var output = new StringWriter();
+            var viewContext = new ViewContext(
+                actionContext,
+                view,
+                new ViewDataDictionary<TModel>(
+                    metadataProvider: new EmptyModelMetadataProvider(),
+                    modelState: new ModelStateDictionary())
+                {
+                    Model = model
+                },
+                new TempDataDictionary(
+                    actionContext.HttpContext,
+                    _tempDataProvider),
+                output,
+                new HtmlHelperOptions());
+
+            await view.RenderAsync(viewContext);
+
+            return output.ToString();
+        }
```

Comment:
I know this is from my commit, but I'm thinking it should be extended to parse the mjml, either automatically or as an option, using [Mjml.AspNetCore](https://www.nuget.org/packages/Mjml.AspNetCore/).

Reply:
I'm trying to decide if we want to commit to Mjml enough to integrate or if it's better to just pre-compile the mjml into html and stick that in our view.  I'm leaning towards the later now -- i'm not sure why we want the complexity or time of spinning up a node instance to compile a view that is the exact same every time instead of just doing it once and checking that code in.  Perhaps that's something we can build automate if this was a larger project, but for now I'd say let's just stick with what we have and we can add mjml custom rendering if we want it later.